epidose/device/update_software_d.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+#
+# Updates software whenever this script is invoked.
+# It is suggested to execute this script after pulling data (git pull)
+# from the production branch.
+#
+# Copyright 2020 Stefanos Georgiou
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+export APP_NAME=update_software
+
+# Pick up utility functions relative to the script's source code
+UTIL="$(dirname "$0")/util.sh"
+
+# Source common functionality (logging, WiFi)
+# shellcheck source=epidose/device/util.sh
+SERVER_URL=none . "$UTIL"
+
+# Wait until an update is required
+wait_till_update_software_needed()
+{
+  # Waiting a process to kill the infinity sleep process
+  sleep infinity &
+  sleep_pid=$!
+  echo "$sleep_pid" > "$SLEEP_UPDATE_SOFTWARE_PID"
+  if ! wait "$sleep_pid" ; then
+    log "Killed during sleep; updating software"
+  fi
+}
+
+# Change all pulled files owner and group to epidose
+# since the git pull was executed as root
+change_owner_and_group()
+{
+  get_git_changes=$(git diff --name-only HEAD@\{0\} HEAD@\{1\})
+  log "Change file(s) owner and group to epidose"
+  for i in $get_git_changes ; do
+    chown epidose:epidose "$i"
+  done
+}
+
+while : ; do
+  wait_till_update_software_needed
+
+  # Go to the local repository to execute the following commands  
+  cd "$LOCAL_REPO"
+  change_owner_and_group
```

Comment:
`chown -R epidose:epidose .` would be enough, no?

Reply:
It would, but currently, it takes around a second to change all files owner:group.
I guess this time is going to increase if we add more files, that is why I created this function.
But, if you insist, I will apply the suggested changes.

Reply:
Then consider `git ls-files | xargs chown epidose:epidose`, which is more robust and efficient.

Reply:
Ok!